Add product as local score

PauBadiaM opened the pull request:
- Added product as local score
- Added unit test for local product
- Fixed small bug in `assert_covered` to handle interactions not present in `adata` nor the resource.

dbdimitrov commented on liana/method/sp/_bivariate_funs.py:
```diff
@@ -144,6 +145,30 @@ def _local_morans(x_mat, y_mat, weight):
+    y_mat = (weight @ y_mat) / total
+
+    # Make features comparable (assumes only +)
+    x_max = np.max(x_mat, axis=0)
```
Maybe absolute so that it also works with negative values?

PauBadiaM commented:
- Added `norm` parameter to `spatial_neighbors`, by default I've set it to `True` but we can discuss it
- Split product score into simple `product` (scores ranging from -inf, +inf) and normalized `norm_product` (scores ranging from -1, +1)
- Updated and added tests for `product` and `norm_product`

I also tested it with the MS data and I can reproduce the results.

dbdimitrov commented on liana/utils/_spatial_neighbors.py:
```diff
@@ -110,6 +113,8 @@ def spatial_neighbors(adata: anndata.AnnData,
         dist.setdiag(0)
     if cutoff is not None:
         dist.data = dist.data * (dist.data > cutoff)
+    if norm:
```
ready to merge - but TBD this when we see each other irl @PauBadiaM
